### ISSUE TITLE
[CI] TransformContinuousIT testContinuousEvents failing

### ISSUE THREAD

**Issue by ywangd** [elastic/elasticsearch#97263]
This failure happened before #93900. The solution (#94184) was to increase the timeout on Windows. But it seems even 60 second is still not sufficient?

**Build scan:**
https://gradle-enterprise.elastic.co/s/b6kea4c4hsslu/tests/:x-pack:plugin:transform:qa:multi-node-tests:javaRestTest/org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT/testContinuousEvents

**Reproduction line:**
```
gradlew ':x-pack:plugin:transform:qa:multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinuousEvents" -Dtests.seed=CC83070B8F857344 -Dtests.locale=en-US -Dtests.timezone=Africa/Dar_es_Salaam -Druntime.java=20
```

**Applicable branches:**
8.8

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT&tests.test=testContinuousEvents

**Failure excerpt:**
```
java.lang.AssertionError: Error in test case [continuous-terms-on-date-pivot-test].If you want to mute the test, please mute [org.elasticsearch.xpack.transform.integration.continuous.TermsOnDateGroupByIT] only, but _not_ [org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT] as a whole.

  at __randomizedtesting.SeedInfo.seed([CC83070B8F857344:342F71CDCB725E50]:0)
  at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.assertForTestCase(TransformContinuousIT.java:539)
  at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$5(TransformContinuousIT.java:474)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1158)
  at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.waitUntilTransformsProcessedNewData(TransformContinuousIT.java:472)
  at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinuousEvents(TransformContinuousIT.java:265)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

  Caused by: java.lang.AssertionError: Timeout [60s] waiting for transform [continuous-terms-on-date-pivot-test] to finish next checkpoint, iteration [8], state [indexing], reason in case of failure [null], last search time [1688085521080]
  Expected: is a value greater than <2023-06-30T00:38:43.248389300Z>
       but: <2023-06-30T00:38:41.080Z> was less than <2023-06-30T00:38:43.248389300Z>

    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
    at org.junit.Assert.assertThat(Assert.java:956)
    at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$4(TransformContinuousIT.java:479)
    at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.assertForTestCase(TransformContinuousIT.java:530)
    at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$5(TransformContinuousIT.java:474)
    at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1158)
    at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.waitUntilTransformsProcessedNewData(TransformContinuousIT.java:472)
    at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinuousEvents(TransformContinuousIT.java:265)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke(Method.java:578)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:1623)

```

**Comment by ywangd** [comment]
There is a new failure on 8.9
https://gradle-enterprise.elastic.co/s/axa6wna63obsw/tests/:x-pack:plugin:transform:qa:multi-node-tests:javaRestTest/org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT/testContinuousEvents?top-execution=1

The exact error message is different
```
Caused by: java.lang.AssertionError: 	
Expected: is "started"	
     but: was "indexing"	
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)	
•••
at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsProcessedNewData$4(TransformContinuousIT.java:495)	
at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.assertForTestCase(TransformContinuousIT.java:530)
```

But since this is the same method and also on Windows, I suspect it might be related to the issue here.

**Comment by andreidan** [comment]
Another one https://gradle-enterprise.elastic.co/s/vufmpfh3y4tco

**Comment by davidkyle** [comment]
Another https://gradle-enterprise.elastic.co/s/vqfsehsuuqxfu/tests/task/:x-pack:plugin:transform:qa:multi-node-tests:javaRestTest/details/org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT/testContinuousEvents?top-execution=1

This time the root cause is a node disconnection
```
Caused by: java.lang.AssertionError: Timeout [60s] waiting for transform [continuous-histogram-pivot-test] to finish next checkpoint, iteration [4], state [failed], reason in case of failure [task encountered more than 0 failures; latest failure: org.elasticsearch.transport.NodeDisconnectedException: [javaRestTest-0][127.0.0.1:41695][indices:data/read/search[phase/query]] disconnected], last search time [1694413742518] |  

Expected: is a value greater than <2023-09-11T06:29:03.813220508Z> |  
  but: <2023-09-11T06:29:02.518Z> was less than <2023-09-11T06:29:03.813220508Z>

```

**Comment by davidkyle** [comment]
`HistogramGroupByIT` and `TermsGroupByIT` tests cases are muted in #99407

**Comment by droberts195** [comment]
Assigning `medium-risk` due to loss of test coverage from muting.

**Comment by piergm** [comment]
antoher instance here:
https://gradle-enterprise.elastic.co/s/uxe4bb3h4alsw

**Comment by cbuescher** [comment]
Another one today on 8.10: https://gradle-enterprise.elastic.co/s/42bzsgb3hfwts

**Comment by mark-vieira** [comment]
Another one today on 8.11: https://gradle-enterprise.elastic.co/s/2j5f2alxeahqs

Looking back 30 days, this *only fails on windows* so that might be something worth taking into consideration. Perhaps it's just the fact that Windows performance is worse so this times out more often on that platform? Maybe we should have more lenient timeouts on that platform?

**Comment by kingherc** [comment]
One more at [elasticsearch / periodic / platform-support / 8.11 / windows-2022 / checkPart2 / platform-support-windows](https://gradle-enterprise.elastic.co/s/kosyhfhb3fj3o)

```
Caused by: java.lang.AssertionError: Timeout [60s] waiting for transform [continuous-terms-on-date-pivot-test] to finish next checkpoint, iteration [8], state [indexing], reason in case of failure [null], last search time [1704438622358]	
Expected: is a value greater than <2024-01-05T07:10:28.916726400Z>	
     but: <2024-01-05T07:10:22.358Z> was less than <2024-01-05T07:10:28.916726400Z>
```

**Comment by przemekwitek** [comment]
> Another one today on 8.11: https://gradle-enterprise.elastic.co/s/2j5f2alxeahqs
> 
> Looking back 30 days, this **only fails on windows** so that might be something worth taking into consideration. Perhaps it's just the fact that Windows performance is worse so this times out more often on that platform? Maybe we should have more lenient timeouts on that platform?

Thanks for the suggestion.
That's what I did in https://github.com/elastic/elasticsearch/pull/104516 .
Hopefully it will help.

**Comment by przemekwitek** [comment]
FTR, https://github.com/elastic/elasticsearch/pull/104516 is now merged in `8.13` and backported to `8.12`.
